lib/compat/wordpress-6.2/get-global-styles-and-settings.php

```diff
@@ -59,160 +59,37 @@ function wp_theme_has_theme_json_clean_cache() {
- *
- * @return string
- */
-function gutenberg_get_global_styles_custom_css() {
-	// Ignore cache when `WP_DEBUG` is enabled, so it doesn't interfere with the theme developers workflow.
-	$can_use_cached = ! WP_DEBUG;
-	$cache_key      = 'gutenberg_get_global_custom_css';
-	$cache_group    = 'theme_json';
-	if ( $can_use_cached ) {
-		$cached = wp_cache_get( $cache_key, $cache_group );
-		if ( $cached ) {
-			return $cached;
-		}
-	}
-
-	if ( ! wp_theme_has_theme_json() ) {
-		return '';
-	}
-
-	$tree       = WP_Theme_JSON_Resolver_Gutenberg::get_merged_data();
-	$stylesheet = $tree->get_custom_css();
-
-	if ( $can_use_cached ) {
-		wp_cache_set( $cache_key, $stylesheet, $cache_group );
-	}
-
-	return $stylesheet;
-}
-
-/**
- * Returns the stylesheet resulting of merging core, theme, and user data.
- *
- * @param array $types Types of styles to load. Optional.
- *                     It accepts as values: 'variables', 'presets', 'styles', 'base-layout-styles.
- *                     If empty, it'll load the following:
- *                     - for themes without theme.json: 'variables', 'presets', 'base-layout-styles'.
- *                     - for themes with theme.json: 'variables', 'presets', 'styles'.
- *
- * @return string Stylesheet.
- */
-function gutenberg_get_global_stylesheet( $types = array() ) {
-	// Ignore cache when `WP_DEBUG` is enabled, so it doesn't interfere with the theme developers workflow.
-	$can_use_cached = empty( $types ) && ! WP_DEBUG;
-	$cache_key      = 'gutenberg_get_global_stylesheet';
-	$cache_group    = 'theme_json';
-	if ( $can_use_cached ) {
-		$cached = wp_cache_get( $cache_key, $cache_group );
-		if ( $cached ) {
-			return $cached;
-		}
-	}
-	$tree                = WP_Theme_JSON_Resolver_Gutenberg::get_merged_data();
-	$supports_theme_json = wp_theme_has_theme_json();
-	if ( empty( $types ) && ! $supports_theme_json ) {
-		$types = array( 'variables', 'presets', 'base-layout-styles' );
-	} elseif ( empty( $types ) ) {
-		$types = array( 'variables', 'presets', 'styles' );
-	}
-
-	/*
-	 * If variables are part of the stylesheet,
-	 * we add them.
-	 *
-	 * This is so themes without a theme.json still work as before 5.9:
-	 * they can override the default presets.
-	 * See https://core.trac.wordpress.org/ticket/54782
-	 */
-	$styles_variables = '';
-	if ( in_array( 'variables', $types, true ) ) {
-		/*
-		 * We only use the default, theme, and custom origins.
-		 * This is because styles for blocks origin are added
-		 * at a later phase (render cycle) so we only render the ones in use.
-		 * @see wp_add_global_styles_for_blocks
-		 */
-		$origins          = array( 'default', 'theme', 'custom' );
-		$styles_variables = $tree->get_stylesheet( array( 'variables' ), $origins );
-		$types            = array_diff( $types, array( 'variables' ) );
-	}
-
-	/*
-	 * For the remaining types (presets, styles), we do consider origins:
+if ( ! function_exists( 'wp_get_global_styles_custom_css' ) ) {
```

Comment:
I was expecting that we'd move some methods from this file to `lib/get-global-styles-and-settings.php`.

As far as I've checked, we should move `gutenberg_get_global_styles_custom_css`, `gutenberg_get_global_stylesheet`, `gutenberg_get_global_settings`, `_gutenberg_clean_theme_json_caches`, `_gutenberg_add_non_persistent_theme_json_cache_group`.

Reply:
I moved everything besides `_gutenberg_add_non_persistent_theme_json_cache_group`. The cache group would be the same, so there's no need to keep that function after 6.2.